scripts/generate-slack-notification-content.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+# Copyright 2024 MongoDB Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+set -euo pipefail
+
+# Check if the parameter is provided
+if [ $# -eq 0 ]; then
+    echo "Usage: $0 <success|failure>"
+    exit 1
+fi
+
+# Check if the parameter is either "success" or "failure"
+if [ "$1" != "success" ] && [ "$1" != "failure" ]; then
+    echo "Invalid parameter. Please provide either 'success' or 'failure'."
```

Comment:
instead of success of failure could we just pass 0 or 1?

Reply:
I think being verbose here is better to understand what's happening in the script. Also, it's easier on the GH action side because ${{ needs.run-test-supported-versions.result }} returns success or failure, so no need to map it to 0 or 1